### PR TITLE
[Fix][Connector-V2][ES]source deserializer error and inappropriate

### LINKED ISSUE #4234: [Bug] [Connector-V2][ES] timestamp type conversion fails and sink stage there will be redundant "
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

timestamp for es
时间戳转换失败

### SeaTunnel Version

2.3.0

### SeaTunnel Config

```conf
env {
  execution.parallelism = 8
  job.mode = "BATCH"
  job.name = "opensearch to console"
}

source {
  Elasticsearch {
    hosts = ["https://xxx:9200"]
    index = "arkime_sessions3*"
    source = ["firstPacket","event.id","event.kind","source.ip","client.nat.ip","source.port","client.nat.port","source.mac","source.hostname","destination.ip","server.nat.ip","destination.port","destination.nat.port","destination.mac","destination.hostname","client.user.name","client.user.domain","client.user.id","host.ip","suricata.alert.severity","rule.name","event.action","event.result","http.response.bytes","event.sequence","network.protocol","network.application","suricata.http.status","suricata.http.http_method","suricata.http.url","suricata.http.hostname","suricata.http.http_user_agent","http.request.referrer","user_agent.device.name","http.authTypeCnt","zeek.smb_cmd.command","file.path","threat.technique.name","related.oui","file.pe.product","suricata.flow.start","event.duration","event.type","host.geo.country_name","host.geo.region_name","host.geo.city_name","source.geo.country_name","source.geo.region_name","source.geo.city_name","source.geo.longitude","source.geo.latitude","destination.geo.country_name","destination.geo.region_name","destination.geo.city_name","destination.geo.longitude","destination.geo.latitude","event.original","network.direction","tags","event.hash"]
    username = "xxx"
    password = "xxx"
    tls_keystore_path = "opensearch.keystroe"
    tls_verify_hostname = false
  }  
}

transform {

}

sink {
    Console {}

}
```

### Running Command

```shell
bin/seatunnel.sh --config ../seatunneljob/job-es-no-transform-to-console.conf -m local
```

### Error Exception

```log
Caused by: java.time.format.DateTimeParseException: Text '1065823369239' could not be parsed at index 0
	at java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:1949)
	at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1851)
	at java.time.LocalDateTime.parse(LocalDateTime.java:492)
	at org.apache.seatunnel.connectors.seatunnel.elasticsearch.serialize.source.DefaultSeaTunnelRowDeserializer.parseDate(DefaultSeaTunnelRowDeserializer.java:209)
	at org.apache.seatunnel.connectors.seatunnel.elasticsearch.serialize.source.DefaultSeaTunnelRowDeserializer.convertValue(DefaultSeaTunnelRowDeserializer.java:161)
	at org.apache.seatunnel.connectors.seatunnel.elasticsearch.serialize.source.DefaultSeaTunnelRowDeserializer.convert(DefaultSeaTunnelRowDeserializer.java:121)
	... 13 more

	at org.apache.seatunnel.engine.client.job.ClientJobProxy.waitForJobComplete(ClientJobProxy.java:118)
	at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:151)
	... 2 more
```

### Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by kpretty**
### for timestamp type data this is wrong
时间解析源代码
The error message is as follows
时间戳转换失败
Therefore, for the timestamp type, it should be parsed into LocalDateTime like this

```java
LocalDateTime.ofInstant(Instant.ofEpochMilli(Long.parseLong(fieldValue)), ZoneId.systemDefault());
```

### string parse inappropriate
It is inappropriate to directly call `toString()` for string type data, and the sink will receive redundant `"`
处理 TextNode 不恰当

### final
The current pr has already dealt with the above problems, and the final effect is as follows
正确处理

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by kpretty** [comment]
e2e for es seems to be problematic

**Comment by TyrantLucifer** [comment]
> e2e for es seems to be problematic

Yup, it seems that this change effect the origin test cases.

**Comment by Hisoka-X** [comment]
Can you fix e2e for this? Thanks

**Comment by kpretty** [comment]
@Hisoka-X ok, I'll fix it in the next pr

**Comment by Hisoka-X** [comment]
> @Hisoka-X ok, I'll fix it in the next pr

ok for me, but we will waiting all ci pass to merge this PR.

**Comment by TyrantLucifer** [comment]
> > @Hisoka-X ok, I'll fix it in the next pr
> 
> ok for me, but we will waiting all ci pass to merge this PR.

CI also has some problems:

image

**Comment by TyrantLucifer** [comment]
@iture123 PTAL.

**Comment by kpretty** [comment]
> > > @Hisoka-X ok, I'll fix it in the next pr
> > 
> > 
> > ok for me, but we will waiting all ci pass to merge this PR.
> 
> CI also has some problems:
> 
> image

Yes, I'm testing locally

**Comment by kpretty** [comment]
@Hisoka-X @TyrantLucifer TBR

**Review by Hisoka-X — DISMISSED** [review]
LGTM

**Comment by TyrantLucifer** [comment]
cc @iture123 PTAL

**Comment by kpretty** [comment]
@TyrantLucifer @Hisoka-X  This conflict is caused by the different ways of processing data types that es is not mapped to. The processing logic of the dev branch throws an exception, but I think this method is not very friendly. Is it better to give a default type user experience

**Comment by EricJoy2048** [comment]
Please fix the conflicts

**Comment by kpretty** [comment]
@EricJoy2048 fixed

File: seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/serialize/source/DefaultSeaTunnelRowDeserializer.java [hunk 1]
```diff
@@ -150,7 +158,8 @@ Object convertValue(SeaTunnelDataType<?> fieldType, String fieldValue)
             LocalDateTime localDateTime = parseDate(fieldValue);
             return localDateTime.toLocalTime();
         } else if (LocalTimeType.LOCAL_DATE_TIME_TYPE.equals(fieldType)) {
```
Comment (iture123) [inline seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/serialize/source/DefaultSeaTunnelRowDeserializer.java]: ES support Multiple date formats, and you should enhance it without removing the old implementation.ES date type can be defined as follows
{
          "type":   "date",
          "format": "yyyy-MM-dd HH:mm:ss||yyyy-MM-dd||epoch_millis"
        }

**Review by iture123 — CHANGES_REQUESTED** [review]

**Comment by kpretty** [comment]
@iture123 there is indeed a problem, I have moved the logic of timestamp processing into `parseDate`

**Review by iture123 — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]

**Review by TyrantLucifer — APPROVED** [review]

**Pull request merged**